web_client/src/textIntervalReader.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ *  A higher order component that reads text from `readTextInterval`: an async function that returns
+ *  `{ text, readingEvery }`, where `text` is a text string read and `readingEvery` is the number
+ *  of seconds after which new `text` should be read. Every `readingEvery` seconds, this component
+ *  will read text from `readTextInterval` and pass the read text to `WrappedComponent`s `text` prop.
+ */
+const withTextIntervalReader = (WrappedComponent, readTextInterval) => (
+    class extends Component {
+        state = {
+            text: ''
+        }
+        onTextIntervalRead({ text }) {
+            this.setState({text});
+        }
+        onFirstTextIntervalRead({ text, readingEvery }) {
+            this.intervalId = setInterval(() => {
+                readTextInterval().then(this.onTextIntervalRead.bind(this));
+            }, readingEvery * 1000);
+            this.onTextIntervalRead({text});
+        }
+        componentDidMount() {
+            readTextInterval()
+                .then(this.onFirstTextIntervalRead.bind(this));
+        }
+        componentWillUnmount() {
+            clearInterval(this.intervalId);
+        }
+        render = () => (
+            <WrappedComponent text={this.state.text} {...this.props} />
+        );
+    }
+);
+
+withTextIntervalReader.propTypes = {
+    WrappedComponent: PropTypes.element.isRequired,
+    readTextInterval: PropTypes.func.isRequired
```

Comment:
What do you mean it "doesn't care about the interval aspect"? It returns `text` and returns the `reading_every` interval.

Reply:
That being said, it could be nice if the signature were

```jsx
const withTextIntervalReader = (WrappedComponent, readText, interval) => (...);

withTextIntervalReader.propTypes = {
    WrappedComponent: PropTypes.element.isRequired,
    readTextInterval: PropTypes.func.isRequired,
    interval: PropTypes.number.isRequired
};
```

but then `App` has to know how to read `readingEvery` / `interval` from the finneganForever api. I do think that the component is much more reusable if it doesn't depend on a function returning `readingEvery` on a text read.

For the time being I think `readTextWithInterval` is a better name.